src/snapred/backend/recipe/FetchGroceriesRecipe.py

```diff
@@ -1,14 +1,22 @@
 from typing import Any, Dict
 
+from snapred.backend.data.LocalDataService import LocalDataService
 from snapred.backend.log.logger import snapredLogger
 from snapred.backend.recipe.algorithm.FetchGroceriesAlgorithm import FetchGroceriesAlgorithm as FetchAlgo
+from snapred.backend.recipe.algorithm.Utensils import Utensils
 from snapred.meta.decorators.Singleton import Singleton
 
 logger = snapredLogger.getLogger(__name__)
 
 
 @Singleton
 class FetchGroceriesRecipe:
+    def __init__(self):
+        # NOTE: workaround, we just add an empty host algorithm.
+        utensils = Utensils()
+        utensils.PyInit()
+        self.mantidSnapper = utensils.mantidSnapper
```

Comment:
since you're import mantid snapper now, remove the import of the FetchAlgo and then replace the use of `algo = FetchAlgo()` etc with something like
``` python
res = self.mantidSnapper.FetchGroceriesAlgorithm(
    Filename=filename,
    OutputWorkspace=workspace,
    LoaderType=loader,
    LoaderArgs=loaderArgs,
    {instrumentPropertySource: instrumentSource},
)
self.mantidSnapper.executeQueue()

data["result"] = True
data["loader"] = res.LoaderType
data["workspace"] = res.OutputWorkspace
```

Reply:
I tried doing this.  Unfortunately it changes thrown exceptions so that several negative tests fail.  Fixing it requires fixing things like MantidSnapper and GenericRecipe and is not worth it for this PR